### DISCUSSION TITLE
Users that login with same gmail account are generating new wallets, can we allow user to select wallet just like with email login?

### DISCUSSION THREAD

**Discussion by chusla** [Web3Auth/Web3Auth#953] [Help]
There was no change to app or verifiers.  same implementation but now users generated a second wallet and app is not recognizing their user account.  This is a pretty big issue when trying to maintain consistent user experience

As a workaround, is there a way to force social media logins to give user option to select wallet if there are multiple wallets?  Just like with torus login or email login?

@shahbaz17 @himanshuchawla009 

Thank you!

**Comment by MontoyaAndres** [comment]
Any help? I'm having the same issue

**Reply by YZhenY** [reply]
Which SDK are y'all using? The torus wallet or web3auth?

Are you guys using more than one? 

An example here would help us help yall

**Reply by chusla** [reply]
Hi thanks for the response we are using the web3auth sdk and logging in via the built in Gmail social media auth method.

Will try to replicate but basically when logging into the app over time I've had a new wallet generated.


This is my app link: demo.one2all.io/version-test/ 

You can login yourself but the first time you log in should work fine.  The problem is this is observed over time.  Maybe we are doing something  unbeknownst to us that is changing the underlying verifier?  Not sure.

**Reply by chusla** [reply]
I've figured out a workaround to force the login to use the right wallet for web3auth which is to use web3auth, choose connect wallet, select torus and then login with my Gmail account in torus and then select the wallet.  But this is moreso a workaround once a second wallet was generated.  What would be better is to understand what potentially causes a second wallet to be generated when the base white listed url has not changed.  Thanks!

**Comment by chaitanyapotti** [comment]
Can you provide a minimal reproducible example or provide more context?

**Comment by MontoyaAndres** [comment]
The issue is when many apps are using web3auth, if I login in app1 with my email andres@gmail.com a unique wallet is going to be generated. If I login to app2 with my exact email, another unique wallet is going to be generated.

So, my question would be, how can we have a unique wallet per user per app? If I login to app1 and app2 with my email andres@gmail.com, I'm going to have the same wallet for both apps. This is what metamask does, where I create a wallet once and I can use it everywhere.

@chaitanyapotti @YZhenY
